R/cache.R

```diff
@@ -116,14 +123,23 @@ moveInstalledPackageToCache <- function(packagePath,
   cachedPackagePath <- cacheLibDir(packageName, hash, packageName)
   backupPackagePath <- tempfile(tmpdir = dirname(cachedPackagePath))
 
+  # helper function to remove the package from its original location and
+  # create a symlink to the cached version.
+  packageSymlink <- function() {
+      unlink(packagePath, recursive = TRUE)
+      symlink(cachedPackagePath, packagePath)
```

Comment:
Can we handle a failed symlink here? Right now if the symlink fails then the application will be busted (the package will just be gone from the private library).

I would prefer something like:

1. Move package installation to temporary path in same directory,
2. Attempt a symlink,
3. On success, remove the directory produced in 1),
4. On failure, move the directory produced in 1) back home.